lib/server.js

```diff
@@ -853,6 +835,91 @@ WalletService.prototype.getMainAddresses = function(opts, cb) {
   });
 };
 
+WalletService.prototype._getLatestActiveAddressIndex = function(addresses, cb) {
+  var self = this;
+  var bc = self._getBlockchainExplorer(addresses[0].network);
+  var activityFound = false;
+  var i = addresses.length;
+  async.whilst(function() {
+    return i > 0 && !activityFound;
+  }, function(next) {
+    var address = addresses[--i].address;
+    activityFound = !!address.hasActivity;
+    if (address.hasActivity) return next();
+
+    bc.getAddressActivity(address, function(err, res) {
+      if (err) return next(err);
+      activityFound = !!res;
+      return next();
+    });
+  }, function(err) {
+    if (err) return cb(err);
+    if (!activityFound) return cb(null, -1);
+
+    var address = addresses[i];
+    address.hasActivity = true;
+    self.storage.storeAddress(address, function(err) {
+      return cb(err, i);
+    });
+  });
+};
+
+/**
+ * Returns the latest unused change address or creates a new one.
+ * @returns {Address} address
+ */
+ WalletService.prototype.getNextChangeAddress = function(cb) {
+   var self = this;
+
+   self._runLocked(cb, function(cb) {
+     self.getWallet({}, function(err, wallet) {
+       if (err) return cb(err);
+       if (!wallet.isComplete()) return cb(Errors.WALLET_NOT_COMPLETE);
+       return self._getChangeAddress(wallet, function(err, address) {
+         return self.storage.storeAddress(address, function(err) {
```

Comment:
storeAddress is called always, even if the address is not new.

Reply:
I am use bitcoin address. Please. Active  
1K9312KKDshshNy43GFwCnkw44zeF4pWW6